apps/emqx/src/emqx_frame.erl

```diff
@@ -739,6 +757,12 @@ parse_will_payload(Bin) when
         expected_bytes => 2
     }).
 
+-compile({inline, [maybe_binary_copy/1]}).
+maybe_binary_copy(Bin) when size(Bin) =< ?SMALL_BINARY ->
```

Comment:
Nit. Not sure if compiler is nowadays smart enough to figure out to specialize `size/1` (which is polymorphic in the argument).
```suggestion
maybe_binary_copy(Bin) when byte_size(Bin) =< ?SMALL_BINARY ->
```